### PR TITLE
Overhaul keyscan to use the raw input message queue

### PR DESCRIPTION
**Pull request by wwebb-r7**
This modifies the current key logging facilities in the stdapi extension to use the raw input api.  In it's current state, the key logging code can miss keystrokes due to polling synchronization issues, such as when someone is typing a memorized password very quickly.  The raw input subsystem queues all messages from input devices at the driver level, ensuring that queries from the userland API will not miss anything.

Unfortunately, this probably breaks support for NT4.  I recommend users needing support for such systems go and download d3adg1rlzz_KeYLoGGa_666.exe from Silicon Toad's Infinity Void.

This will break a couple of framework modules, as key modifiers will no longer need to be processed in the same manner with GetAsyncKeyState().  I'm working on it.

For testing purposes, the keyscan_extract method in lib/rex/post/meterpreter/extensions/stdapi/ui.rb will need to be modified (it basically becomes a nop returning its own argument).  I'll update with the appropriate framework PR when I get to it.

Update:  Associated framework PR necessary for testing is here https://github.com/rapid7/metasploit-framework/pull/8011 - or you could always edit the file yourself

### CODE REVIEW

**Comment by busterb** [comment]
jenkins, add to whitelist

**Comment by bcook-r7** [comment]
jenkins, add to whitelist

**Comment by busterb** [comment]
Given the number of NT 4.0 workstations where a keyboard is connected and being actively used is probably very close to zero, I think this is a very reasonable compromise. The code looks fine so far, will test tomorrow.

File: c/meterpreter/source/extensions/stdapi/server/ui/keyboard.c [hunk 1]
```diff
@@ -40,73 +45,141 @@ DWORD request_ui_enable_keyboard(Remote *remote, Packet *request)
...
+    }
+    
+    // message loop
+    while(GetMessage(&msg, NULL, 0, 0) > 0)
+    {
+        TranslateMessage(&msg);
+        DispatchMessage(&msg);
+    }
+    
+    return msg.wParam;
```
Comment (sempervictus) [inline c/meterpreter/source/extensions/stdapi/server/ui/keyboard.c]: probably want to _return (int)(msg.wParam);_ or change the prototype/def

**Comment by wwebb-r7** [comment]
Closing this one since the direction we're going (Unicode) is significantly different.  I'll have branches for this and framework up by the EoD.

File: c/meterpreter/source/extensions/stdapi/server/ui/keyboard.c (hunk 1 above)
Comment (wwebb-r7) [inline c/meterpreter/source/extensions/stdapi/server/ui/keyboard.c]: Good catch.  Will get updated in my current branch

**Pull request closed without merging**